### ISSUE TITLE
iPhone X screen resolution support?

### ISSUE THREAD

**Issue by beyondthetech** [yoshisuga/MAME4iOS#15]
How difficult is it to implement the new resolution and screen form factor for the iPhone X?

**Comment by yoshisuga** [comment]
I don't have an iPhone X, but I think this project probably needs updating to support the retina resolutions first. I think when I get the next bigger screen iPhone X, I'd be more motivated to fix this :)

**Comment by yoshisuga** [comment]
I have an iPhone XS Max now, and I've started down the path of supporting the larger screen.

I'm not sure if there's that much to gain in terms of screen space used, but it'll be nicer to use the full resolution, I guess?

**Comment by yoshisuga** [comment]
I got started on this:
![img_8808](https://user-images.githubusercontent.com/564774/46498720-f3ebe380-c7b9-11e8-96a6-81c52568142c.PNG)

I added a launch screen storyboard so it'll support any iOS device size, but I need to define frame sizes for each iOS device screen type. I was thinking of doing things 'the right way' by being able to handle any screen size automatically either via autolayout or some other way, but as someone once said, "ain't nobody got time for that". It seems a lot easier to just add the config files for each device type and be mostly done with it.

**Comment by beyondthetech** [comment]
This is what it looks like on my iPhone X🅂 Max.

**Comment by yoshisuga** [comment]
It’s still a work in progress on a branch (modern_screen_sizes). I’ll need to add support for the different iPad sizes. 

The thing that’s more of a pain is the controller layouts. They’re all in the wrong place but they can be moved around freely through settings. I’m planning on fixing the default layouts for each device but I might just put that off til later

**Comment by beyondthetech** [comment]
Awesome, the modern_screen_sizes build is almost fully functional because of the ability to move the controls and buttons, however in landscape mode, none of the buttons are visible (perhaps off screen), so I can’t grab them to move them into position.

**Comment by yoshisuga** [comment]
It's still a work-in-progress. I pushed some changes so you should see the landscape layout on the xs max again.

**Comment by yoshisuga** [comment]
The master branch has been updated with support for iPhone X/XR/XS/XS max and iPad Pro.

When you open the app, the buttons will be misplaced and you will need to rearrange them yourself by going to Option - Settings - Game Input - Change Layout.

I've also removed support for selecting multiple skins for the onscreen controls. Without overhauling the onscreen controls, it's hard to support the different screen sizes that are out there, and I don't think they're that important.
